### PR TITLE
Added example for Material UI/Yup/Dynamic Reload, Reset

### PR DESCRIPTION
**Pull request by tsaockham**
Second try... Hello, took me ages to figure this all out, so I thought I'd share. The main issue I was having was updating the default values when the prop changed, that and dealing with the controlled and uncontrolled component issues. Setting the key value on the Control component handled the issue, however this then confused the validator on reset, so I had to create a custom reset method. Hopefully this will be useful to somebody, or if there is a better way I'd like to know.

### CODE REVIEW

**Comment by bluebill1049** [comment]
thanks a lot @tsaockham for the PR

File: examples/materialUIYupDynamicReloadReset.tsx.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+This example uses contolled componets, with yup valiation
+and can handle changes to the incoming prop. In this case
+the prop that is updated is called "accountvalues".  This
+form can be used for add/edit via the "actionstate" prop.
+This works with Material-UI components and hooks.
+*/
+
+import React, { useState, useEffect } from "react";
+import ReactDOM from "react-dom";
+import clsx from "clsx";
```
Comment (bluebill1049) [inline examples/materialUIYupDynamicReloadReset.tsx.js]: please remove this

File: examples/materialUIYupDynamicReloadReset.tsx.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+import PropTypes from "prop-types";
```
Comment (bluebill1049) [inline examples/materialUIYupDynamicReloadReset.tsx.js]: remove this dep.

File: examples/materialUIYupDynamicReloadReset.tsx.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import { makeStyles } from "@material-ui/styles";
```
Comment (bluebill1049) [inline examples/materialUIYupDynamicReloadReset.tsx.js]: remove this.

File: examples/materialUIYupDynamicReloadReset.tsx.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  CardActions,
+  Divider,
+  Grid,
+  Button,
+  TextField
+} from "@material-ui/core";
+import { useForm, Controller } from "react-hook-form";
+import * as yup from "yup";
+
+const useStyles = makeStyles(theme => ({
```
Comment (bluebill1049) [inline examples/materialUIYupDynamicReloadReset.tsx.js]: remove extra style

**Comment by bluebill1049** [comment]
hmm thanks for the PR, can we please isolate the example to a single issue and just for form, and remove all those other dependencies. take a look other examples.

File: examples/materialUIYupDynamicReloadReset.tsx.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+  const resetForm = () => {
+    Object.keys(accountvalues).forEach(key => {
+      setValue(key, accountvalues[key]);
+    });
+    setSaveDisabled(true);
+  };
+
+  useEffect(() => {
+    setSaveDisabled(true);
+    resetForm();
```
Comment (bluebill1049) [inline examples/materialUIYupDynamicReloadReset.tsx.js]: there is a `reset` api. is that not working for you?

**Comment by vensauro** [comment]
the CSB is broken, need to export the component as default

**Comment by vensauro** [comment]
sorry talk here, where is not my problem, but why is the problem that this example try to solve?  
 i don`t understant this.  
  
[i make some modification](https://codesandbox.io/s/withered-monad-jt2h9) to use the default reset api, and see the behavior to be same, i don`t see something?

**Comment by tsaockham** [comment]
Okay, I've made a new sandbox to show the issue I was trying to solve.  https://codesandbox.io/s/vigilant-mclean-ovywx

To recreate the issue, open the console and play around with the broken and fixed versions.  Note that when you hit submit, the old values come, even though the fields in the form show the new values.  When you do a keypress on any of the fields, that field alone will update, but the other fields will remain as the old ones.  The reset API function visibly fixes the issues, but the data is still the old data.

Also, if you play around with the account options that have no name or no balance, you can see that the validation also does not reset in the broken form.

On the fixed version I loop over the values and run the API setValue against the incoming prop item, this fixes the issue.

Should I pull this PR and submit this as an issue?  Or is it something I'm doing wrong on my end?

Thanks.

**Comment by bluebill1049** [comment]
thanks a lot for the example @tsaockham can we isolate to a simpler version (it is rather complex to understand and reproduce) and submit an issue with it? I am closing this PR and let's discuss in the issue panel.

**Pull request closed without merging**